src/components/data_search/DatasetSearchTable.jsx

```diff
@@ -211,13 +211,18 @@ export const DatasetSearchTable = (props) => {
     getExportableDatasets(datasets);
   });
 
-  useEffect(() => {
-    const fullQuery = assembleFullQuery();
-    try {
+  const searchAndFilter = useRef(
+    _.debounce((fullQuery) => {
       DataSet.searchDatasetIndex(fullQuery).then((filteredDatasets) => {
         const newFiltered = datasets.filter(value => filteredDatasets.some(item => _.isEqual(item, value)));
         setFiltered(newFiltered);
       });
+    }, 250));
```

Comment:
The JIRA says the debounce time should be 150ms, was that too short?

Reply:
no I just didn't see that, thank you!